multinet/db.py

```diff
@@ -382,12 +425,20 @@ def delete_table(workspace: str, table: str) -> str:
     return table
 
 
+def _run_aql_query(aql: AQL, query: str) -> Cursor:
+    try:
+        aql.validate(query)
+        cursor = aql.execute(query)
+    except (AQLQueryValidateError, AQLQueryExecuteError) as e:
+        raise MalformedRequestBody(str(e))
+
+    return cursor
+
+
 def aql_query(workspace: str, query: str) -> Generator[Any, None, None]:
     """Perform an AQL query in the given workspace."""
     aql = get_workspace_db(workspace).aql
```

Comment:
Shouldn't this be called with `readonly = True`?

It makes me wonder if we should flip the polarity of the default value--make `get_workspace_db()` readonly by default, and force us to call it with a flag if we want a writable DB. I know that sounds obnoxious, but it's definitely more fail-safe.

I opened #409 to discuss this.

Reply:
> Shouldn't this be called with readonly = True?

Probably, I wasn't sure if we were going to enforce readonly on general aql queries. If that's the case, I can add it.


> It makes me wonder if we should flip the polarity of the default value--make `get_workspace_db()` readonly by default, and force us to call it with a flag if we want a writable DB. I know that sounds obnoxious, but it's definitely more fail-safe.
> 
> I opened #409 to discuss this.

I'm generally in favor :+1:

Reply:
> > Shouldn't this be called with readonly = True?
> 
> Probably, I wasn't sure if we were going to enforce readonly on general aql queries. If that's the case, I can add it.
> 

Unless I'm missing something, I think we have to: otherwise, anyone with read access could write mutating AQL queries and completely screw up the workspace. The whole purpose of the readonly Arango user was to safely enable AQL queries, right?

Reply:
True, I'll make the change